pkg/apis/eventing/v1/trigger_validation.go

```diff
@@ -64,6 +66,16 @@ func (ts *TriggerSpec) Validate(ctx context.Context) *apis.FieldError {
 				errs = errs.Also(fe)
 			}
 		}
+		if ts.Filter.Expression != "" {
+			_, err := eventfilter.ParseFilterExpr(ts.Filter.Expression)
+			if err != nil {
+				fe := &apis.FieldError{
+					Message: fmt.Sprintf("Invalid filter expression: %q", err),
+					Paths:   []string{"filter.expression"},
+				}
+				errs = errs.Also(fe)
+			}
+		}
```

Comment:
What if I put both `ts.filter.attributes` and `ts.filter.expression`?
I would expect that such trigger spec isn't valid or that we should define a priority. The same comment applies to `v1beta1` validation.

Reply:
Can you add a UT for the validation?

Reply:
There is some ut already for that, if there is both attributes and expression, then both are tested and the filter does an `and` on the two results

Reply:
That seems the most reasonable approach and doesn't break anything

Reply:
Ok, but I would document this behavior.

Reply:
Probably it needs doc both in knative.dev/docs pr I'm going to do and in the crd struct

Reply:
Added doc in the TriggerFilter struct, i'll add more in knative.dev/docs